modules/HomePage/components/TaskList/TaskList.tsx

```diff
@@ -84,23 +101,47 @@ const TaskList: React.FC<TaskListProps> = ({
   };
 
   const handleEditSubmit = (task: Task) => {
-    const updatedTask: Task = {
-      ...task,
-      title: editFormState.title,
-      description: editFormState.description,
-      dueDate: new Date(editFormState.dueDate),
-      priority: editFormState.priority,
+    const priorityMapping: { [key: number]: "HIGH" | "MEDIUM" | "LOW" } = {
+      1: "HIGH",
+      2: "MEDIUM",
+      3: "LOW",
     };
-    saveTask(updatedTask);
-    setEditFormState({
-      title: "",
-      description: "",
-      dueDate: "",
-      priority: "MEDIUM",
-    });
-  };
-  const { data: todos, isLoading } = useGetTodosQuery();
 
+    const updatedTask: Partial<Task> = {
+      title: editFormState.title || task.title,
+      description: editFormState.description || task.description,
+      dueDate: editFormState.dueDate
+        ? new Date(editFormState.dueDate)
+        : task.dueDate,
+      priority:
+        priorityMapping[Number(editFormState.priority)] || task.priority,
+    };
+
+    setLocalTodos((prevTodos) =>
+      prevTodos.map((t) => (t.id === task.id ? { ...t, ...updatedTask } : t))
+    );
+
+    updateTodo({
+      id: task.id,
+      ...updatedTask,
+    })
+      .unwrap()
+      .then(() => {
+        console.log("Task updated successfully");
```

Comment:
I think the user should be notified of this rather than a log. Maybe show a toast notification to the user?

Reply:
Addressed in new commit